### ISSUE TITLE
[v9.0.0] Difference between `#touch` and `#paper_trail.touch_with_version`?

### ISSUE THREAD

**Issue by fukayatsu** [airblade/paper_trail#1073]
Is this an expected behavior?

### my_bug_report.rb

```ruby
# frozen_string_literal: true

# Use this template to report PaperTrail bugs.
# Please include only the minimum code necessary to reproduce your issue.
require "bundler/inline"

# STEP ONE: What versions are you using?
gemfile(true) do
  ruby "2.5.0"
  source "https://rubygems.org"
  gem "activerecord", "5.1.4"
  gem "minitest", "5.10.3"
  gem "paper_trail", "9.0.0", require: false
  gem "sqlite3"
end

require "active_record"
require "minitest/autorun"
require "logger"

# Please use sqlite for your bug reports, if possible.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
ActiveRecord::Base.logger = nil
ActiveRecord::Schema.define do
  # STEP TWO: Define your tables here.
  create_table :users, force: true do |t|
    t.text :first_name, null: false
    t.timestamps null: false
  end

  create_table :versions do |t|
    t.string :item_type, null: false
    t.integer :item_id, null: false
    t.string :event, null: false
    t.string :whodunnit
    t.text :object, limit: 1_073_741_823
    t.text :object_changes, limit: 1_073_741_823
    t.integer :transaction_id
    t.datetime :created_at
  end
  add_index :versions, %i[item_type item_id]
  add_index :versions, [:transaction_id]

  create_table :version_associations do |t|
    t.integer  :version_id
    t.string   :foreign_key_name, null: false
    t.integer  :foreign_key_id
  end
  add_index :version_associations, [:version_id]
  add_index :version_associations, %i[foreign_key_name foreign_key_id],
    name: "index_version_associations_on_foreign_key"
end
ActiveRecord::Base.logger = Logger.new(STDOUT)
require "paper_trail/config"

# STEP THREE: Configure PaperTrail as you would in your initializer
PaperTrail::Config.instance.track_associations = true

require "paper_trail"

# STEP FOUR: Define your AR models here.
class User < ActiveRecord::Base
  has_paper_trail
end

# STEP FIVE: Please write a test that demonstrates your issue.
class BugTest < ActiveSupport::TestCase
  def test_touch_with_version
    user = User.create(first_name: "Jane")
    user.paper_trail.touch_with_version
    assert_equal 'Jane', user.versions.last.reify.first_name
  end

  def test_touch
    user = User.create(first_name: "Jane")
    user.touch
    assert_equal 'Jane', user.versions.last.reify.first_name
  end
end

# STEP SIX: Run this script using `ruby my_bug_report.rb`
```

### result

```
  1) Failure:
BugTest#test_touch [report.rb:77]:
Expected: "Jane"
  Actual: nil

2 runs, 2 assertions, 1 failures, 0 errors, 0 skips
```

**Comment by jaredbeck** [comment]
Maybe @westonganger can answer? See also https://github.com/airblade/paper_trail/pull/1063

**Comment by westonganger** [comment]
Seems like a bug. Maybe related to this? https://github.com/rails/rails/issues/30466

What happens if you change this line:

https://github.com/airblade/paper_trail/blob/master/lib/paper_trail/model_config.rb#L123

to `in_after_callback: false`

**Comment by westonganger** [comment]
Also seems very similar to https://github.com/airblade/paper_trail/issues/1047

**Comment by key88sf** [comment]
Not sure but what you are describing here may be the same as this: https://github.com/airblade/paper_trail/issues/1076

I tried to be very explicit about the difference there.

**Comment by jaredbeck** [comment]
> Not sure but what you are describing here may be the same as this: #1076

Closing via https://github.com/airblade/paper_trail/pull/1078.

**Comment by fukayatsu** [comment]
> What happens if you change this line:
> 
> https://github.com/airblade/paper_trail/blob/master/lib/paper_trail/model_config.rb#L123
> 
> to in_after_callback: false

Thank you, @westonganger!
This seems to fix the problem.
I'll create a PR about it. cc/ @jaredbeck

**Comment by jaredbeck** [comment]
Fixed by #1085, released in 9.0.2.
